test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/blocks/UserTaskBlockBuilder.java

```diff
@@ -0,0 +1,97 @@
+import io.camunda.zeebe.model.bpmn.builder.UserTaskBuilder;
+import io.camunda.zeebe.test.util.bpmn.random.BlockBuilder;
+import io.camunda.zeebe.test.util.bpmn.random.BlockBuilderFactory;
+import io.camunda.zeebe.test.util.bpmn.random.ConstructionContext;
+import io.camunda.zeebe.test.util.bpmn.random.ExecutionPathSegment;
+import io.camunda.zeebe.test.util.bpmn.random.IDGenerator;
+import io.camunda.zeebe.test.util.bpmn.random.RandomProcessGenerator;
+import io.camunda.zeebe.test.util.bpmn.random.steps.AbstractExecutionStep;
+import io.camunda.zeebe.test.util.bpmn.random.steps.StepCompleteUserTask;
+import java.util.Random;
+
+/** Generates a user task */
+public class UserTaskBlockBuilder implements BlockBuilder {
+
+  private final String taskId;
+  private final String boundaryErrorEventId;
+  private final String boundaryTimerEventId;
+  private final String errorCode;
+  private final boolean hasBoundaryEvents;
+  private final boolean hasBoundaryErrorEvent;
+  private final boolean hasBoundaryTimerEvent;
+
+  public UserTaskBlockBuilder(final ConstructionContext context) {
+    final IDGenerator idGenerator = context.getIdGenerator();
+    final Random random = context.getRandom();
+
+    taskId = idGenerator.nextId();
+    boundaryErrorEventId = "boundary_error_" + taskId;
+    boundaryTimerEventId = "boundary_timer_" + taskId;
+    errorCode = "error_" + taskId;
+
+    hasBoundaryErrorEvent =
+        random.nextDouble() < RandomProcessGenerator.PROBABILITY_BOUNDARY_ERROR_EVENT;
+    hasBoundaryTimerEvent =
+        random.nextDouble() < RandomProcessGenerator.PROBABILITY_BOUNDARY_TIMER_EVENT;
+    hasBoundaryEvents = hasBoundaryErrorEvent || hasBoundaryTimerEvent;
+  }
+
+  @Override
+  public AbstractFlowNodeBuilder<?, ?> buildFlowNodes(
+      final AbstractFlowNodeBuilder<?, ?> nodeBuilder) {
+    final UserTaskBuilder taskBuilder = nodeBuilder.userTask(taskId).name(taskId);
+    AbstractFlowNodeBuilder<?, ?> result = taskBuilder;
+
+    if (hasBoundaryEvents) {
+      final BoundaryEventBuilder boundaryEventBuilder =
+          new BoundaryEventBuilder(taskId, taskBuilder);
+
+      if (hasBoundaryErrorEvent) {
+        result = boundaryEventBuilder.connectBoundaryErrorEvent(boundaryErrorEventId, errorCode);
+      }
+
+      if (hasBoundaryTimerEvent) {
+        result = boundaryEventBuilder.connectBoundaryTimerEvent(boundaryTimerEventId);
+      }
+    }
+
+    return result;
+  }
+
+  @Override
+  public ExecutionPathSegment findRandomExecutionPath(final Random random) {
+    final ExecutionPathSegment result = new ExecutionPathSegment();
+    result.appendDirectSuccessor(new StepCompleteUserTask(taskId));
+
+    if (hasBoundaryTimerEvent) {
+      // set an infinite timer as default; this can be overwritten by the execution path chosen
+      result.setVariableDefault(
+          boundaryTimerEventId, AbstractExecutionStep.VIRTUALLY_INFINITE.toString());
+    }
+
+    return result;
+  }
```

Comment:
:x: Currently, a user task can have a timer and/or an error boundary event. But it generates no execution path that triggers the boundary events.

Like in the [JobWorkerTaskBlockBuilder](https://github.com/camunda-cloud/zeebe/blob/develop/test-util/src/main/java/io/camunda/zeebe/test/util/bpmn/random/blocks/JobWorkerTaskBlockBuilder.java#L160-L166), we should generate execution paths for the boundary events.